AnnService/inc/Core/Common/ConcurrentSet.h

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _SPTAG_COMMON_CONCURRENTSET_H_
+#define _SPTAG_COMMON_CONCURRENTSET_H_
+
+#include <shared_mutex>
+#include <unordered_set>
+
+namespace SPTAG
+{
+    namespace COMMON
+    {
+        template <typename T>
+        class ConcurrentSet
+        {
+        public:
+            ConcurrentSet();
+
+            ~ConcurrentSet();
+
+            bool find(const T& key) const;
```

Comment:
if this only returns a Boolean value, consider using `contains` or `exist` as function name